putio.py

```diff
@@ -186,24 +195,40 @@ def _download_directory(self, dest='.', delete_after_download=False):
         if delete_after_download:
             self.delete()
 
-    def _download_file(self, dest='.', delete_after_download=False):
-        response = self.client.request(
-            '/files/%s/download' % self.id, raw=True, stream=True)
+    def _download_file(self, dest='.', delete_after_download=False, range_size=RANGE_SIZE, chunk_size=CHUNK_SIZE):
+        filepath = os.path.join(dest, self.name)
 
-        filename = re.match(
-            'attachment; filename=(.*)',
-            response.headers['content-disposition']).groups()[0]
-        # If file name has spaces, it must have quotes around.
-        filename = filename.strip('"')
+        if os.path.exists(filepath):
+            first_byte = os.path.getsize(filepath)
+        else:
+            first_byte = 0
 
-        with open(os.path.join(dest, filename), 'wb') as f:
-            for chunk in response.iter_content(chunk_size=1024):
-                if chunk:  # filter out keep-alive new chunks
-                    f.write(chunk)
-                    f.flush()
+        logging.debug('file %s is currently %d, should be %d' % (filepath, first_byte, self.size))
 
-        if delete_after_download:
-            self.delete()
+        with open(filepath, 'ab') as f:
+            # Split up file into blocks of RANGE_SIZE each
+            while first_byte < self.size:
+                if first_byte + range_size < self.size:
```

Comment:
I think i've uncovered a pretty serious flaw here. An off by one error (of course). :)